utils/funcexp/funchelpers.h

```diff
@@ -371,7 +371,7 @@ inline uint32_t calc_mysql_week( uint32_t year, uint32_t month, uint32_t day,
     return days / 7 + 1;
 }
 
-inline bool calc_time_diff(int64_t time1, int64_t time2, int l_sign, long long* seconds_out, long long* microseconds_out)
+inline bool calc_time_diff(int64_t time1, int64_t time2, int l_sign, long long* seconds_out, long long* microseconds_out, bool isDateTime = true)
```

Comment:
isDateTime determines whether to treat time1 and time2 as full DateTimes or as Times